Prevent empty author links in Posts list tables

faisal-alvi opened the pull request:
Trac ticket: https://core.trac.wordpress.org/ticket/62913

## Description:

This PR addresses an issue in the **Posts List Table** where posts without an assigned author display an empty author column. A similar fix is already applied to the Media List Table (https://github.com/WordPress/wordpress-develop/pull/8271), but the issue also affects other list tables.

### Changes Made

• Added a fallback text (— no author) when `get_the_author()` returns an empty string.
• Ensured `get_the_author_meta( 'ID' )` is only used when `get_the_author()` is not empty, preventing invalid author links.
• Mirrors the approach taken in the Media List Table fix for consistency.

This aligns with feedback from @joedolson on https://core.trac.wordpress.org/ticket/62913#comment:5, where he suggested extending the fix to all list tables, not just the Media List Table.

## Testing Instructions

Check the comment below.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

faisal-alvi committed:
Prevent empty author links in Posts list tables

faisal-alvi committed:
update the class name in the doc comment

faisal-alvi commented:
## Testing Instructions

1. Log in to the WordPress admin panel.
2.	Navigate to Posts → Add New and create a new post.
3.	Assign an existing user as the post’s author.
4.	Publish the post.
5.	Go to Posts → All Posts.
6.	Verify that the Author column displays the correct author name.
7.	Click on the author name and ensure it links to the correct author archive.
8.	Create a new post and assign it to a test user.
9.	Navigate to Users → All Users and delete the test user.
10.	Choose the option that removes the user but keeps their posts OR use `UPDATE wp_posts SET post_author = 0 WHERE post_type = 'post' LIMIT 1;` query to reset the author.
11.	Go to Posts → All Posts.
12.	Verify that the Author column shows — no author instead of being empty.

audrasjb committed:
Small change just to trigger github actions again

audrasjb committed:
Revert previous commit

qasumitbagthariya commented:
QA Test Report
---

I tested this PR using the following query to reset the author:

`UPDATE wp_posts SET post_author = 0 WHERE post_type = 'post' LIMIT 1;`

After running the query and removing the author from the post, I noticed that a dash (-) sign appears in place of the author name.


**Before**
![image](https://github.com/user-attachments/assets/cb158987-695f-4099-9714-ad3729dfc3e5)


**After**
![image](https://github.com/user-attachments/assets/1e8fe522-d48f-4e6c-9564-cec86fa19d00)


**Testing Environment** 

<Details>

- WordPress: WordPress 6.8-beta1
- PHP: 8.2.
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS 15.2 

</Details>

joedolson committed:
Merge branch 'trunk' into pr/8488

joedolson committed:
Provide message arg in tests with multiple methods

joedolson committed:
Wrap "no author" in parens to match existing strings and test.

joedolson committed:
Fetch table from existing setup method.

joedolson committed:
Update wpPostsListTable.php

joedolson committed:
Simplify assertions.

joedolson committed:
This is just to see what's being created.

joedolson committed:
No need to call get_the_author when iterating the post object already.

joedolson committed:
OK, that doesn't work; revert.

joedolson committed:
Update media column_author method to match post tables.

joedolson committed:
Further simplify assertions.

The purpose of these tests is to identify whether the author column contains appropriate info; the individual parts are not important.
